src/v/utils/mutex.h

```diff
@@ -35,7 +36,7 @@ class mutex {
     using time_point = typename ss::semaphore::time_point;
 
     mutex()
-      : _sem(1) {}
+      : _sem(ssx::make_named_semaphore(1, "mutex")) {}
```

Comment:
I'd be tempted to move `src/v/utils/mutex.h` to `src/v/ssx/mutex.h` - I "own" both, but I plan to eliminate `src/v/utils`, and you'd be doing me a favour. You're touching everything anyway.

Reply:
> I'd be tempted to move src/v/utils/mutex.h to src/v/ssx/semaphore.h - I "own" both, but I plan to eliminate src/v/utils, and you'd be doing me a favour. You're touching everything anyway.

agree. the fate of utils/ is questionable. i think the future is something practical like common/ and ssx/. model/, random/, compression/, bytes/ etc... are all starting to take on common/ like roles and the dependencies are becoming more complicated.

Reply:
Cool. Would be happy to. Would like to do the mutex stuff as a separate followup PR.